test/connectivity/NetworkController/Program.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System;
+    using System.Net.NetworkInformation;
+    using System.Runtime.Loader;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Docker.DotNet;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+        const int DockerNetworkInterfaceNameLenght = 12;
+        const int DockerNetworkInterfaceStartIndex = 0;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            var cts = new CancellationTokenSource();
+            AssemblyLoadContext.Default.Unloading += ctx => cts.Cancel();
+            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
+            var tcs = new TaskCompletionSource<bool>();
+            cts.Token.Register(s => ((TaskCompletionSource<bool>)s).SetResult(true), tcs);
+
+            switch (Settings.Current.NetworkControllerMode)
+            {
+                case NetworkControllerMode.Offline:
+                    await StartOfflineMode();
+                    break;
+                case NetworkControllerMode.Satellite:
+                    await StartSatelliteMode();
+                    break;
+            }
+
+            await tcs.Task;
+            return 0;
+        }
+
+        static Task StartSatelliteMode() => throw new NotImplementedException();
+
+        static async Task StartOfflineMode()
+        {
+            var delay = Settings.Current.StartAfter;
+            var networkInterfaceName = GetDockerInterfaceName();
+            IOfflineController offlineController;
+
+            if (networkInterfaceName.HasValue)
+            {
+                await networkInterfaceName.ForEachAsync(
+                    async name =>
+                    {
+                        offlineController = new NetworkInterfaceOfflineController(name);
+
+                        foreach (Frequency item in Settings.Current.Frequency)
+                        {
+                            await RunScheduledTask(delay, offlineController, item.OfflineFrequency, item.OnlineFrequency, item.RunsCount);
```

Comment:
nit: offlineController is not needed.
```suggestion
                            await RunScheduledTask(delay, new NetworkInterfaceOfflineController(name), item.OfflineFrequency, item.OnlineFrequency, item.RunsCount);
```

Reply:
I changed the code structure here, it needs to create all controllers to make sure it clean everything before starting

---
In reply to: [353901818](https://github.com/Azure/iotedge/pull/2040#discussion_r353901818) [](ancestors = 353901818)